src/logging.rs

```diff
@@ -1,12 +1,40 @@
-pub fn init() {
-    let guard = sentry::init((
-        "https://aa37d7fba870e93bf72c2138b1029d35@o1.ingest.us.sentry.io/4507408549347328",
-        sentry::ClientOptions {
+use std::{borrow::Cow, str::FromStr};
+
+use sentry::types::Dsn;
+
+use crate::config::Config;
+
+#[derive(Debug)]
+pub struct LoggingConfig {
+    /// The DSN key to use for sentry error reporting. If left empty sentry will not be
+    /// initialized.
+    pub sentry_dsn: Option<String>,
+
+    /// The sentry environment to report errors to.
+    pub sentry_env: Option<Cow<'static, str>>,
+}
+
+impl From<Config> for LoggingConfig {
+    fn from(config: Config) -> Self {
+        Self {
+            sentry_dsn: config.sentry_dsn,
+            sentry_env: config.sentry_env,
+        }
+    }
+}
```

Comment:
I believe this is a misuse of `From`. `A: From<B>` is for cases where `B` can be converted to `A` faithfully and losslessly, which is not the case here (I mean, it's the case now, but it won't be once `Config` gets bigger). You have to throw parts of `Config` away to make it fit into a `LoggingConfig`. See https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from for what I mean.

A better way to do this would be to give `Config` a field `logging: LoggingConfig`.

Reply:
> can be converted to A faithfully and losslessly

Ah I didn't know this ok.

Reply:
Okay, switched this over to a method on the `LoggingConfig` called `from_config`

Reply:
You could also make `logging::init` just take two individual parameters, unless you expect `LoggingConfig` to grow

Reply:
ah yeah.maybe that's easier lol. I'll adjust later I think